File: C#/BluffinMuffin.Protocol/Lobby/RegisteredMode/CreateUserCommand.cs
using BluffinMuffin.Protocol.DataTypes.Attributes;
using BluffinMuffin.Protocol.Enums;

namespace BluffinMuffin.Protocol.Lobby.RegisteredMode
{
    /// <summary>
    /// This commands will create a new user in the database.
    /// </summary>
    public class CreateUserCommand : AbstractLobbyCommand, ICommandWithResponse<CreateUserResponse>
    {
        /// <summary>
        /// The username you want to have
        /// </summary>
        [ExampleValue("ericmas001")]
        public string Username { get; set; }
        /// <summary>
        /// The password you want to use
        /// </summary>
        [ExampleValue("0nc3Up0nAT1m3")]
        public string Password { get; set; }
        /// <summary>
        /// An email address to reach this user
        /// </summary>
        [ExampleValue("ericmas001@hotmail.com")]
        public string Email { get; set; }
        /// <summary>
        /// The display name you want for this user
        /// </summary>
        [ExampleValue("Sponge Bob")]
        public string DisplayName { get; set; }

        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public CreateUserResponse Response(bool success, BluffinMessageId msgId, string message)
        {
            return new CreateUserResponse(this) { Success = success, MessageId = msgId, Message = message };
        }
    }
}

File: Python/BluffinMuffin.Protocol/lobby/registered_mode/create_user_command.py
from abstract_lobby_command import AbstractLobbyCommand

__author__ = 'ericmas001@gmail.com'



class CreateUserCommand(AbstractLobbyCommand):
    def __init__(self, obj):
        super().__init__(obj)
        self.username = obj['Username']
        self.password = obj['Password']
        self.email = obj['Email']
        self.display_name = obj['DisplayName']

    def __str__( self ):
        return '{0} ({1} : {2}, {3}, {4})'.format(super().__str__(), self.username, self.password, self.email, self.display_name)
